Sources/General/KingfisherManager.swift

```diff
@@ -342,7 +342,19 @@
             result in
             handler(currentSource: source, result: result)
         }
-
+    }
+    
+    public func performDownloadIfNecessary(source: Source) {
+        downloadIfNotCached(with: source)
+    }
+    
+    private func downloadIfNotCached(with source: Source) {
+        let cached = cache.isCached(forKey: source.cacheKey)
+        if cached {
+            downloader.downloadImage(with: source.url!, options: [])
+        } else {
+            // Do nothing since already cached.
+        }
```

Comment:
**Suggestion:** The `performDownloadIfNecessary` method is public but it just calls the private `downloadIfNotCached` method without any additional logic. Consider removing the `performDownloadIfNecessary` method and make the `downloadIfNotCached` method public instead. [enhancement]
```suggestion
    public func downloadIfNotCached(with source: Source) {
        let cached = cache.isCached(forKey: source.cacheKey)
        if !cached {
            downloader.downloadImage(with: source.url!, options: [])
        } else {
            // Do nothing since already cached.
        }
    }
```